plugins/distance_table.hpp

```diff
@@ -104,11 +106,44 @@ template <class DataFacadeT> class DistanceTablePlugin final : public BasePlugin
             const int bearing = input_bearings.size() > 0 ? input_bearings[i].first : 0;
             const int range = input_bearings.size() > 0 ? (input_bearings[i].second?*input_bearings[i].second:10) : 180;
             facade->IncrementalFindPhantomNodeForCoordinate(route_parameters.coordinates[i],
-                                                            phantom_node_vector[i], 1, bearing, range);
+                                                            phantom_node_vector[i], 1, bearing, range, 1100, 0);
+
+            if (phantom_node_vector[i].size() == 1)
+            {
+                at_least_one_only_big_component |= !phantom_node_vector[i].back().component.is_tiny;
+            }
 
             BOOST_ASSERT(phantom_node_vector[i].front().is_valid(facade->GetNumberOfNodes()));
         }
 
+        // If we have one phantom node that only exists in a big component, then we will force all others
```

Comment:
Why not use the logic from `viaroute`?

1. Check if phantom nodes are all in the same component (doesn't matter if small or big)
2. If they are not try to use the big component for each node

I think it should be consistent.